### PR TITLE
Fix the graph in EnsoGL stats monitor panel to synchronize FPS with other stats

### PR DESCRIPTION
**Pull request by akavel**
### Pull Request Description

Fix EnsoGL runtime statistics collection and processing, so that all values in a single vertical slice of the Monitor Panel graph (accessible through `control-option-tilde` shortcut) represent stats for the same animation/rendering frame.

Before the change, for any vertical slice of the graph, the FPS value was shown for the previous (`i - 1`) rendering frame, whereas the other values were shown for the current (`i`) rendering frame.

The alignment of the stats will become even more important in the future, when the values will be included in more detailed profiling reports, which is planned.

#### Visuals

The GUI that is impacted (if in a subtle way) by the changed code is the performance monitor panel, accessible with a `ctrl-option-tilde` keyboard shortcut in the IDE.

A screenshot of the Monitor panel **after** applying the PR, taken in IDE built with `./run dist`:

Screenshot 2022-02-04 at 14 01 47

A video recording of the panel taken in the same IDE (note that the process of recording itself has an impact on the FPS):

https://user-images.githubusercontent.com/273837/152533944-3876f7a1-8fa1-4da2-9865-4813dd945442.mov

A screenshot of the Monitor panel **before** applying the PR, taken in IDE built with `./run dist`:

Screenshot 2022-02-02 at 12 38 09

A video recording of the panel taken in the same IDE (note that the process of recording itself has an impact on the FPS):

https://user-images.githubusercontent.com/273837/152157638-06d4e7fc-70ab-4f3e-be86-d4f7e2a327d8.mov

### Important Notes

https://www.pivotaltracker.com/story/show/181140499

[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/monitor.rs]: All comments should be valid sentences.

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/monitor.rs]: Please check the comments with gramarly (as described in the handbook). For example, "earlier frame before" is a sentence that uses 2 words to describe the same thing.

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/monitor.rs]: If you want to refer to Rust value, use the Rust doc syntax for it please, like ``` [`Correct`] ```.

File: lib/rust/ensogl/core/src/display/world/stats.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/world/stats.rs]: In such a case, this API should be changed to use turbo fish syntax instead of doing defaults everywhere. Passing things as values make sense if there is any case when the values change. In case the values do not change, pass them as types instead.

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/monitor.rs]: could you make in the macro some params default? Like e.g. if the last param is not passed, use `None`. This way you might be able to create all samplers as one-line expressions :)

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (akavel) [reply lib/rust/ensogl/core/src/debug/monitor.rs]: As per our separate discussion on chat, I improved the names of the local variables and removed the comments.

File: lib/rust/ensogl/core/src/debug/monitor.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/monitor.rs]: why not just `let panel = Panel::new(self.config.clone(), S::default());`?

**Review by wdanilo — APPROVED** [review]
Amazing! ❤️ 
Thank you so much for the screenshots and movies – were extremely helpful to validate that nothing bad happened.

**Comment by akavel** [comment]
(*Process note:* the code-review is now accepted, therefore the PR is waiting for Review by @MichaelMauderer on Task https://www.pivotaltracker.com/story/show/181140499 before merge.)

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/181140499">Fix the graph in EnsoGL stats monitor panel to synchronize FPS with other stats - Pivotal Tracker</a></strong></div></blockquote>

**Pull request merged**
